### ISSUE TITLE
windows getting AttributeError: module 'os' has no attribute 'getuid'. Did you mean: 'getpid'?

### ISSUE THREAD

**Issue by borick** [OpenDevin/OpenDevin#169]
**Describe the bug**
getting the error AttributeError: module 'os' has no attribute 'getuid'. Did you mean: 'getpid'? on windows
updated to latest in the main branch

**Steps to Reproduce**
1. run  uvicorn opendevin.server.listen:app --port 3000
2. cd frontend;  npm run start -- --port 3001
3. wait for UI to load get the error in the uvicorn shell - "

**Expected behavior**
OpenDevin works

**Actual behavior**

PS C:\Users\boris\thirdparty\OpenDevin> uvicorn opendevin.server.listen:app --port 3000
INFO:     Started server process [111408]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:3000 (Press CTRL+C to quit)
INFO:     ('127.0.0.1', 54050) - "WebSocket /ws" [accepted]
INFO:     connection open
Task exception was never retrieved
future: <Task finished name='create controller' coro=<Session.create_controller() done, defined at C:\Users\boris\thirdparty\OpenDevin\opendevin\server\session.py:114> exception=AttributeError("module 'os' has no attribute 'getuid'")>
Traceback (most recent call last):
  File "C:\Users\boris\thirdparty\OpenDevin\opendevin\server\session.py", line 132, in create_controller
    self.controller = AgentController(self.agent, directory, callbacks=[self.on_agent_event])
  File "C:\Users\boris\thirdparty\OpenDevin\opendevin\controller\__init__.py", line 33, in __init__
    self.command_manager = CommandManager(workdir)
  File "C:\Users\boris\thirdparty\OpenDevin\opendevin\controller\command_manager.py", line 24, in __init__
    self.shell = DockerInteractive(id="default", workspace_dir=dir)
  File "C:\Users\boris\thirdparty\OpenDevin\opendevin\sandbox\sandbox.py", line 51, in __init__
    uid = os.getuid()
AttributeError: module 'os' has no attribute 'getuid'. Did you mean: 'getpid'?

**Additional context**
Windows in PowerShell

**Comment by rbren** [comment]
Thanks for the report! I think we've mainly been running on mac.

Can you try this branch: `rb/fix-uid`?

**Comment by rbren** [comment]
Proposed fix: https://github.com/OpenDevin/OpenDevin/pull/170

**Comment by borick** [comment]
Yeeah actually i started up! I think that fix is done!

but there's another issue:

Finish cleaning up Docker container
Exception ignored in atexit callback: <bound method DockerInteractive.cleanup of <opendevin.sandbox.sandbox.DockerInteractive object at 0x0000016C36CE1EA0>>
Traceback (most recent call last):
  File "C:\Users\boris\thirdparty\OpenDevin\opendevin\sandbox\sandbox.py", line 148, in cleanup
    self.container.remove(force=True)
  File "C:\Users\boris\AppData\Local\Programs\Python\Python310\lib\site-packages\docker\models\containers.py", line 352, in remove
    return self.client.api.remove_container(self.id, **kwargs)
  File "C:\Users\boris\AppData\Local\Programs\Python\Python310\lib\site-packages\docker\utils\decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "C:\Users\boris\AppData\Local\Programs\Python\Python310\lib\site-packages\docker\api\container.py", line 1027, in remove_container
    self._raise_for_status(res)
  File "C:\Users\boris\AppData\Local\Programs\Python\Python310\lib\site-packages\docker\api\client.py", line 270, in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
  File "C:\Users\boris\AppData\Local\Programs\Python\Python310\lib\site-packages\docker\errors.py", line 39, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation) from e
docker.errors.NotFound: 404 Client Error for http+docker://localnpipe/v1.43/containers/7ee7554df78ae5b7f15627088f44abd478485eac5aa4cb15bc91a059ac2a87ab?v=False&link

should I raise that as a separate bug?

**Comment by rbren** [comment]
I have a fix for this one already--it's an error that pops up after the program exits.

Did you quit opendevin to make this appear? Are there any other logs?

**Comment by borick** [comment]
> I have a fix for this one already--it's an error that pops up after the program exits.
> 
> Did you quit opendevin to make this appear? Are there any other logs?

i don't think so, I just started it up per the docs... it "activated the agent" - I tried to send that message and received the error (pretty sure...)

**Comment by rbren** [comment]
If you manage to capture it again, I'm hoping there are some more logs above `Finish cleaning up Docker container`

**Comment by borick** [comment]
ok right this time it actually ran for a bit(not sure what I did differently) but it errored with this:

PS C:\Users\boris\thirdparty\OpenDevin> uvicorn opendevin.server.listen:app --port 3000
INFO:     Started server process [45628]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:3000 (Press CTRL+C to quit)
INFO:     ('127.0.0.1', 49661) - "WebSocket /ws" [accepted]
INFO:     connection open
STEP 0
ACTION CmdRunAction(command='ls', background=False)
---
OBSERVATION CONTRIBUTING.md
LICENSE
OpenDevinLogo.jpg
README.md
agenthub
dev_config
evaluation
frontend
opendevin
requirements.txt
workspace

CONTRIBUTING.md
LICENSE
OpenDevinLogo.jpg
README.md
agenthub
dev_config
evaluation
frontend
opendevin
requirements.txt
workspace

==============
STEP 1
ACTION AgentThinkAction(thought='Before starting to build the roguelike, I should check if the README.md file contains any relevant information about the existing project structure or any specific instructions.', runnable=False)
---
ACTION NOT EXECUTABLE
OBSERVATION

==============
STEP 2
ACTION FileReadAction(path='README.md', base_path='')
FileReadAction(path='README.md', base_path='.')
---
Error in loop 'charmap' codec can't decode byte 0x90 in position 1928: character maps to <undefined>

**Comment by borick** [comment]
ok great now this just has to be merged into main :D I raised another issue for the other issue - here - https://github.com/OpenDevin/OpenDevin/issues/178
